x-pack/plugins/reporting/server/lib/store/store.ts

```diff
@@ -109,8 +110,7 @@ export class ReportingStore {
       const client = await this.getClient();
       this.ilmPolicyManager = IlmPolicyManager.create({ client });
     }
-
-    return this.ilmPolicyManager;
+    return this.config.statefulSettings.enabled ? this.ilmPolicyManager : null;
```

Comment:
1. We can rely on making this method **not** called at all from `start()` based on the config (see my feedback in `start()`).
2. The Store class doesn't need a `this.ilmPolicyManager`

```typescript
  private async getIlmPolicyManager() {
    const client = await this.getClient();
    return IlmPolicyManager.create({ client });
  }
```

Reply:
Way better thanks